tests/worker_deployment_test.go

```diff
@@ -2483,12 +2472,17 @@ func (s *WorkerDeploymentSuite) startAndValidateWorkerDeployments(
 		}
 
 		for _, expectedDeploymentSummary := range expectedDeploymentSummaries {
-			deploymentSummaryValidated := false
+			deploymentSummaryFound := false
 			for _, actualDeploymentSummary := range actualDeploymentSummaries {
-				deploymentSummaryValidated = deploymentSummaryValidated ||
-					s.verifyWorkerDeploymentSummary(expectedDeploymentSummary, actualDeploymentSummary)
+				// our assumption is that deployment summaries with the same name are fully ready for checks. May not be true since visibility might take time to update other fields.
```

Comment:
```suggestion
				// Our assumption is that deployment summaries with the same name are fully ready for checks.
				// May not be true since visibility might take time to update other fields.
```

nit: readability and punctuation